### ISSUE TITLE
Add a `--fail-on-deprecated` option to `istioctl validate`

### ISSUE THREAD

**Issue by Stono** [istio/istio#27938]
**Describe the feature request**
We use `istioctl` in our pipelines (and our local cli wrapper) to prevent deploy time failures.  It would be really great if we could make `istioctl` fail when a deprecated resource is detected, to force users to update to the newer version.

For example `authentication.istio.io/v1alpha1` was deprecated and replaced with `peerauthentications.security.istio.io/v1beta1`.  It'd have been great if `istioctl validate --fail-on-deprecated` existed, as it would have forced the update to happen in our pipelines way before the 1.6 upgrade we just did (where the crd was removed thus breaking a bunch of applications - admittedly we missed the upgrade notes around this, I'd just like to have more automated processes to protect against the human error).

**Describe alternatives you've considered**
None, open to suggestions 

[x] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

**Additional context**

**Comment by howardjohn** [comment]
We also recently added warnings. I think deprecation should be exposed as warnings, then we should make this `--fail-on-warning`. Super useful I think

**Comment by howardjohn** [comment]
`istioctl analyze` has a `--failure-threshold` which would do similar

@esnible I think `validate` is a subset of analyze, do we intend to keep both around? Is validate deprecated in favor of analyze? Should it be?

**Comment by esnible** [comment]
We should deprecate `istioctl validate`.

I don't think `--fail-on-deprecated` should be added.  We have `--suppress <code>=<resource>`.  If we need this we should probably handle it the same way, with `--only IST0002=*` or similar.

**Comment by Stono** [comment]
@esnible sorry I'm not familiar with the syntax, would that enable me to exit with a non-zero status code when deprecated apis are detected?

**Comment by esnible** [comment]
@Stono We don't have a good way to suppress all-but one type.

As a work-around, until we have a design and merged implementation, `istioctl analyze | grep IST0002` works in scripts.  It does a bit of extra work, and it sets `$?` to the opposite of what you want, but it can be used in scripting.

**Comment by Stono** [comment]
@esnible just to be clear I'm not looking to suppress all but one type, I'm happy for all warnings to come through, I was just looking for a way to set at which level we would fail?

Of course I could parse the output from istioctl analyze with some custom scripts, however what I was requesting is quite a common pattern with any sort of linting or validation tool, you set a failure threshold or failure codes and the tool itself will exit none-zero when those conditions are met.

**Comment by howardjohn** [comment]
`--failure-threshold` is already there, takes info/warn/error. Then if there is some check that is false positives you can filter it out with `--suppress`.

What @esnible is suggesting, which I think may be useful but not required, is an allowlist option rather than denylist of `--suppress`

**Comment by Stono** [comment]
Ahhh so I could do `istioctl analyze --failure-threshold=warn` and `IST0002` = deprecated warning?

**Comment by Stono** [comment]
> We should deprecate istioctl validate.

This seems to make sense to me by the way so long as we're able to continue to do offline validation with analyze, I don't see any reasons validate exists.

**Comment by Stono** [comment]
Can you see what i'm doing wrong here?:

Deprecated API:
```
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: release-name
  labels:
    app: release-name
    heritage: "Tiller"
    release: "release-name"
    group: delivery-platform
  annotations:
    'atcloud.io/locality-routing': 'true'
spec:
  host: app
  trafficPolicy:
    outlierDetection:
      interval: 5s
      consecutiveGatewayErrors: 5
      consecutive5xxErrors: 0
      baseEjectionTime: 30s
      minHealthPercent: 50
      maxEjectionPercent: 25
    # Note: These values also get modified by shippr-cli during deployments
    loadBalancer:
      localityLbSetting:
        enabled: false
      simple: RANDOM
    connectionPool:
      http:
        h2UpgradePolicy: DO_NOT_UPGRADE
        http1MaxPendingRequests: 2147483646
        http2MaxRequests: 2147483646
        maxRequestsPerConnection: 2147483646
        maxRetries: 2147483646
        idleTimeout: 30m
      tcp:
        maxConnections: 2147483646
        tcpKeepalive:
          time: 30s
          interval: 30s
    tls:
      mode: "ISTIO_MUTUAL"
```

```
❯ istioctl analyze --use-kube=false example.yaml 
✔ No validation issues found when analyzing namespace: search-solr.
```

**Comment by esnible** [comment]
@howardjohn Are networking.istio.io/v1alpha3 DestinationRules deprecated?  It does not say so in https://istio.io/latest/docs/reference/config/networking/destination-rule/

@Stono @jasonwzm I looked into how we might implement a warning for v1alpha3 DRs as in Stono's example.  What I discovered is that the validation framework is internally translating v1beta1 resources into v1alpha3 resources encountered on the mesh for the purpose of validating them.  I can't easily distinguish the old from the new, this makes issuing a deprecation warning impossible.

What's worse, when the CRs are passed in as files, the back-translation isn't done.  https://github.com/istio/istio/issues/28719
